rebasehelper/application.py

```diff
@@ -517,24 +519,25 @@ def build_packages(self):
                         deleted_files = '\n'.join(files['deleted'])
                         logger.warning('Removed files packaged in SPEC file:\n%s', deleted_files)
                     else:
-                        raise RebaseHelperError("Build failed, but no issues were found in the build log %s" % build_log)
+                        if self.conf.build_retries == 0:
+                            raise RebaseHelperError("Build failed, but no issues were found in the build log %s", build_log)
                     self.rebase_spec_file.modify_spec_files_section(files)
 
                 if not self.conf.non_interactive:
-                    if failed_before:
                         msg = 'Do you want rebase-helper to try build the packages one more time'
                         if not ConsoleHelper.get_message(msg):
                             raise KeyboardInterrupt
                 else:
                     logger.warning('Some patches were not successfully applied')
+                #  build just failed, otherwise we would break out of the while loop
+                logger.debug('Number of retries is %s', self.conf.build_retries)
+                if os.path.exists(os.path.join(results_dir, 'RPM')):
                     shutil.rmtree(os.path.join(results_dir, 'RPM'))
+                if os.path.exists(os.path.join(results_dir, 'SRPM')):
                     shutil.rmtree(os.path.join(results_dir, 'SRPM'))
-                    return False
-                #  build just failed, otherwise we would break out of the while loop
-                failed_before = True
-
-                shutil.rmtree(os.path.join(results_dir, 'RPM'))
-                shutil.rmtree(os.path.join(results_dir, 'SRPM'))
+                self.conf.build_retries -= 1
+        if self.conf.build_retries == 0:
```

Comment:
If the build fails also after configured number of retries, we should raise RebaseHelperError. I don't see any reason why returning a value from this function would be beneficial or necessary. Can you please explain?